src/test/resources/features/resolve/uses_and_modules_name_resolve.feature

```diff
@@ -27,27 +27,43 @@ Feature: Resolve use and module name to its declaration
     When I put the caret at NAME as
     Then reference should resolve to "NAME" in "foo.smk"
 
-  Scenario: Refer to other file
+  Scenario Outline: Refer to rules, declared in other sections/modules
     Given a snakemake project
     And a file "boo.smk" with text
     """
+    include: "zoo.smk"
+
+    use rule rule_from_zoo as updated_zoo with: threads: 3
+
     rule z:
       input: "data_file.csv"
     """
+    And a file "zoo.smk" with text
+    """
+    rule zoo_rule: threads: 1
```

Comment:
It is better when a test checks a single idea/feature. Here we test several things together - resolve to module and to a file included in module. Could you please split test in several independent tests that better reflect which case is being tested.